core/src/main/scala/kafka/security/authorizer/AclAuthorizer.scala

```diff
@@ -550,6 +660,31 @@ class AclAuthorizer extends Authorizer with Logging {
   }
 
   private def updateCache(resource: ResourcePattern, versionedAcls: VersionedAcls): Unit = {
+    val currentAces: Set[AccessControlEntry] = aclCache.get(resource) match {
```

Comment:
commit 230ee36b9147a11d7ce299aa9fcbb590324faf68
Added the authorizeByResourceType() API to the benchmark and simulate the worst case: every allow ACL on the same resource has a dominant deny ACL.
Adjust the `resourceCount` parameter to "10000", "40000", "80000" since each cluster is unlikely to have more than 10k resources. Also, since we are testing against the worst case mentioned above, I think the "10000" cases are adequate for us.
Performance result here: https://paste.ubuntu.com/p/k8kV3w6fvF/

Reply:
See below